### ISSUE TITLE
Invalid inline comment behavior

### ISSUE THREAD

**Issue by dantezhu** [tomtom/tcomment_vim#300]
### Env
- gvim v9.0
- tcomment latest

### Steps to reproduce problem:

1. prepare text line like below:
```
abcdefg
```

2. type `v` to enter visual mode, and select `ef`, like below:
![image](https://user-images.githubusercontent.com/1105212/181866472-25d3dea1-58f2-4447-88f6-83e4c223a35b.png)

3. type `gc`

4. the text will changed to:
```
abcd/* efg */
```
but the right result should be:
```
abcd/* ef */g
```

**Comment by tomtom** [comment]
>     the text will changed to:
> 
> |abcd/* efg */ |
> 
> but the right result should be:
> 
> |abcd/* ef */g |

You're right of course. The reason for this is that (within the approach 
of tcomment) there didn't seem to be a reliable way to distingish 
between the last and the last but one position in a line. Maybe 
vim/vimscript have changed in the meantime to allow distinguishing 
between these two situations without having to change too much in the 
plugin code. If you know of a way to handle this consistently across 
operating systems, while taking into account all possible option values, 
please let me know.

**Comment by dantezhu** [comment]
@tomtom got it, tks

**Comment by tomtom** [comment]
One more remark: (for this reason) tcomment also supports "comment
everything to the right" (<c-_>r), which should do what you want to achieve.
